rpc/mux_broker.go

```diff
@@ -95,9 +94,12 @@ func (m *muxBroker) Dial(id uint32) (net.Conn, error) {
 	return stream, nil
 }
 
-// NextId returns a unique ID to use next.
+// NextId returns a unique ID to use next. There is no need for seeding the
+// rand package as the returned ID's aren't stored or used anywhere outside
+// the current runtime. So it's perfectly fine to get the same pseudo-random
+// numbers each time terraform is running.
 func (m *muxBroker) NextId() uint32 {
-	return atomic.AddUint32(&m.nextId, 1)
+	return rand.Uint32()
```

Comment:
I'll open an issue there to discuss this, but I can imagine use cases where this check would actually make sense. For Terraform it doesn't as the channels aren't usually used for more then a couple of minutes...

So would something like this be a workaround until (if) the logic in yamux can be changed?

```go
func (m *muxBroker) NextId() uint32 {
  if m.nextid == math.MaxUint32-2 {
    m.nextid = 0
  }
  return atomic.AddUint32(&m.nextId, 1)
}
```

Maybe still a little dirty, but this would do the trick right?